ports/mimxrt/Makefile

```diff
@@ -467,7 +474,13 @@ $(BUILD)/lib/tinyusb/src/device/usbd.o: CFLAGS += -Wno-missing-braces
 # Build targets
 # =============================================================================
 
+# Do not build the .uf2 target for MIMXRT1050_EVK, MIMXRT1170_EVK, Teensy40 and Teensy41
+# These boards do not support the bootloader properly.
+ifneq ($(BOARD), $(filter $(BOARD), MIMXRT1050_EVK TEENSY40 TEENSY41 MIMXRT1170_EVK))
```

Comment:
I suggest adding a make variable to those boards that use the bootloader, instead of testing for specific boards.  Eg `USE_BOOTLOADER = 1`, and then here `ifeq ($(USE_BOOTLOADER),1)`.

Reply:
OK. I'll do. The default would be to use the bootloader.

Reply:
Done.

Reply:
This switch only controls the creation of the .uf2 file. The memory layout is the same for boards which do not use the uf2 bootloader. That reminds me of adding the rule to add the .uf2 file to the download page. I assume that a .uf2 file is silently skipped if missing.